RedHatInsights/compliance-frontend
# feat(RHINENG-13595): Migrate the rules table on the system details page

Implements https://issues.redhat.com/browse/RHINENG-13595.

This migrates the policy tabs and the rule (rule results) table on the system details view to use REST API endpoints.

Known issues (will be implemented/fixed in a separate PR):
- The tree table view is not yet implemented, only the rows table view must be available for now
- The Remediate button is kept disabled, we require items selection to be improved first

## How to test

Verify that all the reporting policies for the systems are reflected in a separate tab. The rules table should correctly filter, sort and paginate items. All the cells (name, policy, severity, rule state, remediation) must be correctly displayed. Also, check that the rule details are expandable.

## Screenshots

<img width="1511" alt="image" src="https://github.com/user-attachments/assets/8d17127e-f5f0-4351-aa48-d4a1f0283f4b">

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
